pkg/odo/cli/component/create.go

```diff
@@ -660,6 +661,30 @@ func (co *CreateOptions) Run() (err error) {
 	} else {
 		log.Italic("\nPlease use `odo push` command to create the component with source deployed")
 	}
+	if log.IsJSON() {
+		var componentDesc component.Component
+		co.Context, co.LocalConfigInfo, err = genericclioptions.UpdatedContext(co.Context)
+		if err != nil {
+			return err
+		}
+		existsInCluster, err := component.Exists(co.Context.Client, *co.componentSettings.Name, co.Context.Application)
+		if err != nil {
+			return err
+		}
+		if existsInCluster {
+			componentDesc, err = component.GetComponent(co.Context.Client, *co.componentSettings.Name, co.Context.Application, co.Context.Project)
+			if err != nil {
+				return err
+			}
+		} else {
+			componentDesc, err = component.GetComponentFromConfig(co.LocalConfigInfo)
+			if err != nil {
+				return err
+			}
+		}
+		componentDesc.Spec.Ports = co.LocalConfigInfo.GetPorts()
```

Comment:
Why is this needed to be done, shouldn’t the port be populated in the spec?

Reply:
I took what is there in `odo describe` logic. Must be a reason why its there